src/docfx/lib/schema/JsonSchemaValidator.cs

```diff
@@ -262,6 +264,70 @@ private void ValidateProperties(JsonSchema schema, string propertyPath, JObject
             }
         }
 
+        private static void ValidateMaxItemsWhen(JsonSchema schema, string name, JArray array, List<Error> errors)
+        {
+            foreach (var (condition, value) in schema.MaxItemsWhen)
+            {
+                var count = 0;
+                for (var i = 0; i < array.Count; i++)
+                {
+                    if (SchemaMatches(condition, array[i]))
+                    {
+                        count++;
+                    }
+                }
+
+                if (count > value)
+                {
+                    errors.Add(Errors.JsonSchema.ArrayMatchInvalid(
+                        JsonUtility.GetSourceInfo(array),
+                        name,
+                        $"The array must not have more than {value} matched item(s)."));
+                }
+            }
+        }
+
+        private static void ValidateMinItemsWhen(JsonSchema schema, string name, JArray array, List<Error> errors)
+        {
+            foreach (var (condition, value) in schema.MinItemsWhen)
+            {
+                var count = 0;
+                for (var i = 0; i < array.Count; i++)
+                {
+                    if (SchemaMatches(condition, array[i]))
+                    {
+                        count++;
+                    }
+                }
+
+                if (count < value)
+                {
+                    errors.Add(Errors.JsonSchema.ArrayMatchInvalid(
+                        JsonUtility.GetSourceInfo(array),
+                        name,
+                        $"The array must have least {value} matched item(s)."));
+                }
+            }
+        }
+
+        private static bool SchemaMatches(JsonSchema schema, JToken map)
+        {
+            if (map is JObject obj)
+            {
+                foreach (var prop in schema.Properties)
+                {
+                    if (obj.TryGetValue(prop.Key, StringComparison.OrdinalIgnoreCase, out var value)
+                        && prop.Value != null
+                        && prop.Value.Const != null
+                        && prop.Value.Const.Equals(value))
+                    {
+                        return true;
+                    }
+                }
+            }
+            return false;
```

Comment:
How about we recalling validate function here and checking whether there exist new appearing error to indicate match or not.

This modification will bring the supporting for pattern, type, const and so on, for example, the schema we can use
"minItemsWhen": [
                  {
                    "value": 1,
                    "condition": {"properties": {"isCorrect": {"pattern": "^a.*"}}}
                  }
                ],

Reply:
Very good suggestion, I have updated the code. Thanks @xichan-ms